services/src/main/java/io/scalecube/services/Microservices.java

```diff
@@ -0,0 +1,248 @@
+ *        // print the greeting.
+ *         System.<b><font color="9b0d9b">out</font></b>.println(result);
+ *       } else {
+ *         // print the greeting.
+ *         System.<b><font color="9b0d9b">out</font></b>.println(ex);
+ *       }
+ *     });
+ * }
+ * </pre>
+ */
+
+public class Microservices {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(Microservices.class);
+  private static final ServiceProcessor serviceProcessor = new AnnotationServiceProcessor();
+
+  private final ICluster cluster;
+
+  private final IServiceRegistry serviceRegistry;
+
+  private final ServiceProxyFactory proxyFactory;
+
+  private Microservices(ICluster cluster, Object[] services, boolean isSeed) {
+    this.cluster = cluster;
+    this.serviceRegistry = new ServiceRegistry(cluster, services, serviceProcessor, isSeed);
+    this.proxyFactory = new ServiceProxyFactory(serviceRegistry, serviceProcessor);
+    new ServiceDispatcher(cluster, serviceRegistry);
+  }
+
+  public ICluster cluster() {
+    return this.cluster;
+  }
+
+  public void unregisterService(Object serviceObject) {
+    serviceRegistry.unregisterService(serviceObject);
+  }
+
+  private <T> T createProxy(Class<T> serviceInterface, Class<? extends Router> router, Duration timeout) {
+    return proxyFactory.createProxy(serviceInterface, router, timeout);
+  }
+
+  public Collection<ServiceInstance> services() {
+    return serviceRegistry.services();
+  }
+
+  public static final class Builder {
+
+    private Integer port = null;
+    private Address[] seeds;
+    private Object[] services = new Object[0];
+
+    /**
+     * microsrrvices instance builder.
+     * 
+     * @return Microservices instance.
+     */
+    public Microservices build() {
+      ClusterConfig cfg = getClusterConfig();
+      return new Microservices(Cluster.joinAwait(cfg), services, seeds == null);
+    }
+
+    private ClusterConfig getClusterConfig() {
+      Map<String, String> metadata = new HashMap<>();
+
+      if (services.length > 0) {
+        metadata = Microservices.metadata(services);
+      }
+
+      ClusterConfig cfg;
+      if (port != null && seeds != null) {
+        cfg = ConfigAssist.create(port, seeds, metadata);
+      } else if (seeds != null) {
+        cfg = ConfigAssist.create(seeds, metadata);
+      } else if (port != null) {
+        cfg = ConfigAssist.create(port, metadata);
+      } else {
+        cfg = ConfigAssist.create(metadata);
+      }
+      return cfg;
+    }
+
+    public Builder port(int port) {
+      this.port = port;
+      return this;
+    }
+
+    public Builder seeds(Address... seeds) {
+      this.seeds = seeds;
+      return this;
+    }
+
+    /**
+     * Services list to be registered.
+     * 
+     * @param services list of instances decorated with @Service
+     * @return builder.
+     */
+    public Builder services(Object... services) {
+      if (services == null) {
+        this.services = new Object[0];
```

Comment:
one case if someone call builder services null
but its also possible this node is just a proxy and services not called at all.
i still can gain proxy for the services from this node

you can see in the examples